File: solutions/ruby/run_solution.rb
require 'colorize'
require './day_of_the_programmer'

#input parsing and reading
input00 = File.open("../../test-cases/input/input00.txt")
input01 = File.open("../../test-cases/input/input01.txt")
input60 = File.open("../../test-cases/input/input60.txt")

data = input00.map(&:chomp)
year00 = data[0].to_i

data = input01.map(&:chomp)
year01 = data[0].to_i

data = input60.map(&:chomp)
year60 = data[0].to_i


# problem application
start = Time.now
response00 = DayOfTheProgrammer.run year00
response01 = DayOfTheProgrammer.run year01
response60 = DayOfTheProgrammer.run year60
finish = Time.now

puts("solved all the test cases in " + (1000 * (finish - start)).round(2).to_s + "ms")

#expected outputs

output00File = File.open("../../test-cases/output/output00.txt")
output01File = File.open("../../test-cases/output/output01.txt")
output60File = File.open("../../test-cases/output/output60.txt")


data = output00File.map(&:chomp)
output00 = data[0]

data = output01File.map(&:chomp)
output01 = data[0]

data = output60File.map(&:chomp)
output60 = data[0]

if response00 === output00
	puts "INPUT 00".green
else
	puts "INPUT 00".red
end

if response01 === output01
	puts "INPUT 01".green
else
	puts "INPUT 01".red
end

if response60 === output60
	puts "INPUT 60".green
else
	puts "INPUT 60".red
end

File: solutions/java/RunSolution.java
import java.io.*; 
import java.util.Arrays;

public class RunSolution{

	public static final String ANSI_RED = "\u001B[31m";
	public static final String ANSI_GREEN = "\u001B[32m";
	public static final String ANSI_RESET = "\u001B[0m";

	public static void main(String[] args) throws Exception{

		//input parsing and reading
		File input00File = new File("../../test-cases/input/input00.txt");
		BufferedReader brInput00 = new BufferedReader(new FileReader(input00File));

		File input01File = new File("../../test-cases/input/input01.txt");
		BufferedReader brInput01 = new BufferedReader(new FileReader(input01File));

		File input60File = new File("../../test-cases/input/input60.txt");
		BufferedReader brInput60 = new BufferedReader(new FileReader(input60File));

		String st;
		String lineSplit[];

		int year00 = Integer.parseInt(brInput00.readLine());

		int year01 = Integer.parseInt(brInput01.readLine());

		int year60 = Integer.parseInt(brInput60.readLine());
		

		// problem application

		String response00 = DayOfTheProgrammer.run(year00);
		String response01 = DayOfTheProgrammer.run(year01);
		String response60 = DayOfTheProgrammer.run(year60);
		

		// expected outputs

		File output00File = new File("../../test-cases/output/output00.txt");
		BufferedReader brOutput00 = new BufferedReader(new FileReader(output00File));

		File output01File = new File("../../test-cases/output/output01.txt");
		BufferedReader brOutput01 = new BufferedReader(new FileReader(output01File));

		File output60File = new File("../../test-cases/output/output60.txt");
		BufferedReader brOutput60 = new BufferedReader(new FileReader(output60File));


		String output00 = brOutput00.readLine();
		String output01 = brOutput01.readLine();
		String output60 = brOutput60.readLine();

		System.out.println(((response00.equals(output00)) ? ANSI_GREEN : ANSI_RED) + "INPUT 00");
		System.out.println(((response01.equals(output01)) ? ANSI_GREEN : ANSI_RED) + "INPUT 05");
		System.out.println(((response60.equals(output60)) ? ANSI_GREEN : ANSI_RED) + "INPUT 60" + ANSI_RESET);
	}
}